### ISSUE TITLE
Customize Window Title Bar like macOS Default Terminal

### ISSUE THREAD

**Issue by VicBoss1984** [kovidgoyal/kitty#7407]
Hi Everyone,

Does the Kitty terminal allow the user to customize the window title bar beyond including the present-working-directory on macOS? My intention is to emulate the window title bar setup of the default macOS terminal in Kitty, as I am trying to migrate from the macOS terminal to Kitty. 

Here is what the default macOS terminal looks like with the setup that I have for it:

<img width="1337" alt="Screenshot 2024-05-02 at 06 31 38" src="https://github.com/kovidgoyal/kitty/assets/85129181/d26f15b8-2cd4-4106-b09b-06385008e62f">

Here is what the Kitty terminal looks like with the setup that I have for it:

<img width="1392" alt="Screenshot 2024-05-02 at 06 34 21" src="https://github.com/kovidgoyal/kitty/assets/85129181/ed724bb1-c5ae-44fd-9136-bb11f4c61592">

Basically, I am curious if Kitty supports a similar window title customization feature like the macOS terminal, like so:

<img width="780" alt="Screenshot 2024-05-02 at 06 34 50" src="https://github.com/kovidgoyal/kitty/assets/85129181/994fcaab-5e7d-4369-ac51-0fd7dd16f65d">

Many thanks,
Ali

**Comment by kovidgoyal** [comment]
The window title can show anything you like. Configure your shell to set
the window title using the set title escape codes. Once you do so, all
terminals will show whatever that is in their title bars.
